File: Data Structures & Algorithms/sliding-window-maximum/submission-2.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        heap = []
        for i in range(k):
            heapq.heappush(heap, (-nums[i], i))
        out = [heap[0][0] * -1]

        for i in range(k, len(nums)):
            heapq.heappush(heap, (-nums[i], i))
            while len(heap) > 0 and heap[0][1] <= i - k:
                heapq.heappop(heap)
            out.append(heap[0][0] * -1)
        return out

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-2.java
class Solution {
    public boolean canPartition(int[] nums) {
        int total = 0;
        for (int i = 0; i < nums.length; i++){
            total += nums[i];
        }

        if (total % 2 == 0) {
            boolean[] ith_total = new boolean[total / 2 + 1];
            ith_total[0] = true;
            for (int i = 0; i < nums.length; i++) {
                for (int j = ith_total.length; j >= 0; j--) {
                    if (j + nums[i] < ith_total.length && ith_total[j]) {
                        ith_total[j + nums[i]] = true;
                    }
                }
            }
            return ith_total[ith_total.length - 1];
        }
        return false;
    }
}
